### PR TITLE
Add JobDatum GetLogs option

### PR DESCRIPTION
**Pull request by acohen4**

### CODE REVIEW

**Review by jrockway — APPROVED** [review]

File: src/logs/logs.proto [hunk 1]
```diff
@@ -72,6 +74,11 @@ message PipelineJobLogQuery {
   string job = 2;
 }
 
+message JobDatumLogQuery {
+  string job = 1;
```
Comment (robert-uhl) [inline src/logs/logs.proto:78]: Should this be a PPS Job message (i.e., pipeline + job ID) or a job string?

File: src/server/logs/cmds/cmds.go [hunk 2]
```diff
@@ -166,25 +192,31 @@ func Cmds(pachCtx *config.Context, pachctlCfg *pachctl.Config) []*cobra.Command
...
 			case cmd.Flag("project").Changed:
-				if cmd.Flag("datum").Changed || cmd.Flag("app").Changed {
+				if hasDisallowedFlags(cmd, "datum", "app") {
 					fmt.Fprintln(os.Stderr, "only one of [--logQL | --project PROJECT --pipeline PIPELINE | --datum DATUM] may be set")
 					os.Exit(1)
 				}
 				addProjectRequest(req, project)
+			case cmd.Flag("datum").Changed && cmd.Flag("job").Changed:
+				if hasDisallowedFlags(cmd, "app") {
+					fmt.Fprintln(os.Stderr, "only one of [--logQL | --project PROJECT --pipeline PIPELINE | --datum DATUM] may be set")
```
Comment (robert-uhl) [inline src/server/logs/cmds/cmds.go:214]: Need to update the error message (which should probably be a `const` …) to `only one of [--logQL | --project PROJECT --pipeline PIPELINE | --datum DATUM [--job JOB]] may be set"`

File: src/server/logs/cmds/cmds_test.go [hunk 3]
```diff
@@ -75,35 +75,36 @@ func TestGetLogs_default_noauth(t *testing.T) {
 	).Run())
 }
 
-func TestGetLogs_default_nonadmin(t *testing.T) {
-	if testing.Short() {
-		t.Skip("Skipping integration tests in short mode")
-	}
+// TODO: need to replace this test with real loki that can filter by data for specific pipelines and non-admin users
```
Comment (robert-uhl) [inline src/server/logs/cmds/cmds_test.go:78]: Were these tests failing with the changes? I think that it’s probably better to keep them until we replace them, even if they’re not great.

But if they were incorrectly failing … delete away!

File: src/logs/logs.proto (hunk 1 above)
Comment (robert-uhl) [reply src/logs/logs.proto:78]: Notably, the existing pipeline query doesn’t use a PPS pipeline message, but is structured into project name and pipeline name, while the existing job query is structured into a pipeline (= project name and pipeline name) and job ID. Maybe JobDatumLogQuery should similarly use a PipelineJobLogQuery?

Comment (robert-uhl) [reply src/logs/logs.proto:78]: Ah, makes sense because we can uniquely identify with the datum & job IDs together — no need for the pipeline-job ID.

**Review by bbonenfant — APPROVED** [review]

**Pull request merged**